src/odemis/driver/smaract.py

```diff
@@ -3173,3 +3174,1244 @@ def SA_CTL_Move(self, handle, ch, pos_pm, _):
+        Additionally, the parameters from the previously saved referencing procedure are loaded.
+        This function is blocking.
+        """
+        # Loading the configuration will cause the state to become temporarily unstable. Once the process is
+        # finished, the state will be stable again.
+        self.EnableEventNotification(SA_PS_STABLE_STATE_CHANGED_EVENT)
+        self.SetProperty_i32(SA_PS_AF_ADJUSTMENT_RESULT_LOAD_PROP, SA_SI_ENABLED)
+        self._wait_for_progress_event(SA_SI_ENABLED, timeout=600)  # this can take a long time
+        self.DisableEventNotification(SA_PS_STABLE_STATE_CHANGED_EVENT)
+
+    def _save_configuration(self):
+        """
+        Saves the configuration parameters (cf _load_configuration) and internal parameters of the
+        referencing procedure. This function is blocking.
+        """
+        self.SetProperty_i32(SA_PS_AF_ADJUSTMENT_RESULT_SAVE_PROP, SA_SI_ENABLED)
+        # There is no documentation on loading/saving the configuration in the manual, even though
+        # the functions are available in the header file. It is not clear how much time saving the
+        # parameters takes. It can be assumed that it's fast --> wait 1 s.
+        time.sleep(1)
+
+    def _wait_for_progress_event(self, end_state, timeout=float("inf")):
+        """
+        Blocks until progress event is triggered or timeout.
+        It is assumed that only one event type is enabled. The function will wait until the
+        .devEventParameter contains the end_state. This can mean different things depending on the event.
+        end_state (int): state of event variable to wait for
+        timeout (float): maximum time to wait in s. If inf, it will wait forever.
+        raises
+            TimeoutError: timeout exceeded
+            CancelledError: function was cancelled by SA_SI_Cancel
+            SA_SI_Error: other problem reported by device
+        """
+        if timeout == float("inf"):
+            tend = None
+        else:
+            tend = time.time() + timeout
+
+        state = None
+        while state != end_state:
+            if tend is not None:
+                t = tend - time.time()
+                if t < 0:
+                    raise TimeoutError("Timeout limit of %s s exceeded." % timeout)
+            else:
+                t = float("inf")
+
+            ev = self.WaitForEvent(t)  # raise TimeOut/CancelledError by itself
+            if ev.type == SA_PS_AF_ADJUSTMENT_PROGRESS_EVENT:
+                # lowest 16-bits is the adjustment state
+                state = ev.devEventParameter & 0xffff
+            elif ev.type == SA_PS_STABLE_STATE_CHANGED_EVENT:
+                # new state, 0 means unstable, 1 means stable
+                state = ev.devEventParameter
+            elif ev.type == SA_SI_STREAM_ABORTED_EVENT:
+                raise SA_SIError(ev.error, self.core.err_code[ev.error])
```

Comment:
You *might* want to raise a CancelledError in some cases... but let's check, if it's ever needed anyway.